Release workflow

rajdchak opened the pull request:
## Description of change
This workflow automates the release process

#### Relevant issues
<!-- Please add issue numbers. -->
<!-- Please also link them to this PR. -->

#### Does this contribution introduce any breaking changes to the existing APIs or behaviors?
<!-- Please explain why this was necessary. -->

#### Does this contribution introduce any new public APIs or behaviors?
<!-- Please describe them and explain what scenarios they target.  -->

#### How was the contribution tested?
Ran the workflow manually https://github.com/rajdchak/analytics-accelerator-s3/actions/runs/14910238538 

#### Does this contribution need a changelog entry?
- [ ] I have updated the CHANGELOG or README if appropriate

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

SanjayMarreddi reviewed (approved):
Thanks for this!
Please make sure we have added the secrets and variables on this repository as well and update our internal document on the new release process:
- How to trigger this workflow
- Clarify what next steps to be done after this workflow run finishes successfully.

rajdchak commented:
> Thanks for this! Please make sure we have added the secrets and variables on this repository as well and update our internal document on the new release process:
> 
> * How to trigger this workflow
> * Clarify what next steps to be done after this workflow run finishes successfully.

Yes I have already updated the secrets and variables in this repo. Will update the doc soon.

stubz151 commented:
Do we see this release on sonahub? and we checked that the signing passed there? Also have we tested the released jar there and made sure it's bundled normally? (Just trying to think of the problems we ran into the first time we did this on catalogs side)
